azure/windows-webapp/windows-webapp/main.tf

```diff
@@ -0,0 +1,371 @@
+##########
+# Insights
+##########
+
+resource "azurerm_log_analytics_workspace" "main" {
+  count = var.create_application_insights ? 1 : 0
+
+  name                = "log-${var.name}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  tags                = var.tags
+
+  sku               = "PerGB2018"
+  retention_in_days = 30
+}
+
+resource "azurerm_application_insights" "main" {
+  count = var.create_application_insights ? 1 : 0
+
+  name                = "appi-${var.name}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  tags                = var.tags
+
+  application_type = "web"
+  workspace_id     = one(azurerm_log_analytics_workspace.main[*].id)
+}
+
+###############
+# Service Plan
+###############
+
+resource "azurerm_service_plan" "main" {
+  count = var.plan["create"] ? 1 : 0
+
+  name                = var.plan["name"] == null ? "asp-${var.name}" : var.plan["name"]
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  tags                = var.tags
+
+  sku_name = var.plan["sku_name"]
+
+  os_type                = "Windows"
+  zone_balancing_enabled = var.plan["zone_balancing"]
+}
+
+locals {
+  plan_id = coalesce(one(azurerm_service_plan.main[*].id), var.plan["id"])
```

Comment:
Is it worth moving this to locals.tf ?

Reply:
Kept this in line with the linux-webapp module and how that's written. May be worth factoring that in for a new PR to update both webapps as they're both similar constructs?

Reply:
Added this here for future work: https://github.com/orgs/ans-coe/projects/1/views/1?pane=issue&itemId=26670765